src/solver/conv_asm_implicit_gemm_bwd_v4r1_dynamic.cpp

```diff
@@ -0,0 +1,202 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2020 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include "miopen/solver.hpp"
+#include "miopen/handle.hpp"
+#include <miopen/conv/invokers/impl_gemm_dynamic.hpp>
+#include <miopen/generic_search.hpp>
+#include <miopen/gcn_asm_utils.hpp>
+#include "implicitgemm_util.hpp"
+
+namespace miopen {
+namespace solver {
+
+static inline bool FindImplicitGemmDynamicKernelBwd(const ConvolutionContext& ctx,
+                                                    std::string& kernel_name,
+                                                    int& block_size,
+                                                    int& grid_size)
+{
+    // TODO: add more dynamic kernel to expand support range, and update this function
+    // clang-format off
+    // refer to ConvolutionContextInterpreter, in bwd most dimension is reversed
+    int hi          = ctx.out_height;
```

Comment:
It is better to call ConvolutionContextInterpreter e.g. hi = ConvolutionContextInterpreter::GetInputHeightHi

https://github.com/ROCmSoftwarePlatform/MIOpen/blob/16d87c1952e41b54b36fe6cb59a2a40cc3275934/src/solver/implicitgemm_util.hpp#L131

Reply:
ok